htdocs/ecommerceng/class/business/eCommerceSynchro.class.php

```diff
@@ -1557,7 +1507,7 @@ public function synchProduct($toNb=0)
                     dol_syslog("- Process synch of product remote_id=".$productArray['remote_id']);
 
                     $counter++;
-                    if ($toNb > 0 && $counter > $toNb) break;
+                   // if ($toNb > 0 && $counter > $toNb) break;
```

Comment:
Why commenting this ?
If synchProduct is called with a toNb value defined (for example 100), it means we want to synch only toNb record. If you disabled this, this will sync all and not only oldest 100.

Reply:
We already recover the number of product desired here line 1430
$products = $this->eCommerceRemoteAccess->convertRemoteObjectIntoDolibarrProduct($resulttoupdate, $toNb);	// Return max $toNb record only
And for Woocommerce i get $toNb products (+ their variations products (get for each product ^^')).